server/dbSchemas/attendanceSchema.js

```diff
@@ -0,0 +1,38 @@
+const mongoose = require('mongoose');
+const { Schema } = mongoose;
+
+const userSchema = new Schema({
+    email: { type: String, required: true, trim: true },
+    name: { type: String, required: true, trim: true },
+    password: { type: String, required: true },
+    is_student: { type: Boolean, required: true },
+    is_professor: { type: Boolean, required: true },
```

Comment:
Since each user is either prof or student we only need one boolean. For example if is_professor is true we know it is a professor, and if it is false we know it is a student.
```suggestion
    is_professor: { type: Boolean, required: true },
```